text/0389-dynamic-tag-names.md

```diff
@@ -43,14 +43,29 @@ With the `element` helper proposed in this RFC, this can be accomplished with so
 
 ## Detailed design
 
-We propose to add a new `element` helper that takes a tag name and generates a contextual component that, when invoked, renders the selected element.
+We propose to add a new `element` helper that takes a single positional argument.
+
+* When passed a non-empty string it generates a contextual component that, when invoked, renders an element with the same tag name as the passed string, along with the passed attributes (if any), modifiers (if any) and yields to given block (if any).
+* When passed an empty string, it generates a contextual compoment that, when invoked, yields to the given block without wrapping it an element and ignores any passed modifiers and attributes.
+* When passed `null` or `undefined`, it will return `null`.
```

Comment:
The proof-of-concept addon currently throws here. The motivation for this is to mostly match the `(component)` helper's behavior with this, and acknowledge that there are useful patterns arising from this. Note that when `null` is "invoked" it's a no-op, which is why this works for components.